.github/workflows/prepare-release.yml

```diff
@@ -0,0 +1,45 @@
+name: SmallRye Prepare Release
+
+on:
+  pull_request:
+    types: [ closed ]
+    paths:
+      - '.github/project.yml'
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}
+  cancel-in-progress: true
+
+permissions:
+  contents: write
+
+jobs:
+  prepare-gradle:
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v4
+
+      - name: git author
+        run: |
+          git config --global user.name "SmallRye CI"
+          git config --global user.email "smallrye@googlegroups.com"
+      
+      - uses: radcortez/project-metadata-action@main
+        name: retrieve project metadata
+        id: metadata
+        with:
+          github-token: ${{secrets.GITHUB_TOKEN}}
+          metadata-file-path: '.github/project.yml'
+
+      - name: Prepare Gradle
+          echo "version=${{steps.metadata.outputs.current-version}}" > tools/gradle-plugin/gradle.properties
+          git add tools/gradle-plugin/gradle.properties
+          git commit -m "Update Gradle plugin version"
```

Comment:
You need to push the changes somewhere, the `prepare-release` job below won't execute in the same context as this job

Reply:
I'd recommend having a hook to perform the changes you need in the prepare-release.yml, like in https://github.com/quarkiverse/.github/blob/main/.github/workflows/prepare-release.yml#L118-L124

Reply:
Ah, we need to do a `git push` there.

Then, the job below would just check out the new commit.

Reply:
For `git push` to work, it depends on the ruleset configured in this repository too (I don't know if the current user is able to bypass the branch restrictions). In any case, I think it would be better if it used the GitHub App token, as in 
https://github.com/smallrye/.github/blob/main/.github/workflows/prepare-release.yml#L35-L40